### DISCUSSION TITLE
Local rules not working?

### DISCUSSION THREAD

**Discussion by Lesterol** [Security-Onion-Solutions/securityonion#12284] [Unsupported Versions] [answered]
Hello everyone. I've been experimenting all day with adding local rules, but haven't gotten any positive results.
I'm using version 2.4.40. I added rules according to the documentation via Administration –> Configuration –> idstools –> rules –> Local Rules.
Even adding the simplest rules, such as 
```
alert icmp any any -> any any (msg:"ICMP Testing Rule"; sid:1000001; rev:1;)
alert tcp any any -> any any (msg:"TCP Testing Rule"; sid:1000002; rev:1;)
alert udp any any -> any any (msg:"UDP Testing Rule"; sid:1000003; rev:1;)
alert icmp any any -> $HOME_NET any (msg:"ICMP Test"; sid 1000004; rev:1;)
```
I still don’t see local alerts, but only default ones from all.rules. I even see events in Hunt and the traffic matches these local rules, but there is no alert. Can anyone help?
Standalone verison

**Comment by reyesj2** [comment, marked as answer]
Looks like your last rule is missing ```:``` after the sid
```alert icmp any any -> $HOME_NET any (msg:"ICMP Test"; sid:1000004; rev:1;)```

I went and added those rules to my idstools -> rules -> local rules
```
alert icmp any any -> any any (msg:"ICMP Testing Rule"; sid:1000001; rev:1;)
alert tcp any any -> any any (msg:"TCP Testing Rule"; sid:1000002; rev:1;)
alert udp any any -> any any (msg:"UDP Testing Rule"; sid:1000003; rev:1;)
alert icmp any any -> $HOME_NET any (msg:"ICMP Test"; sid:1000004; rev:1;)
```
Then clicked the 'Synchronize grid' button at the top of SOC and I started to get alerts for the new rules (after about 15 minutes for the grid sync to complete)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/94730068/a1a3996b-29f9-4526-a035-7fe26e39cf4c)

**Comment by Lesterol** [comment]
> Looks like your last rule is missing `:` after the sid `alert icmp any any -> $HOME_NET any (msg:"ICMP Test"; sid:1000004; rev:1;)`
> 
> I went and added those rules to my idstools -> rules -> local rules
> 
> ```
> alert icmp any any -> any any (msg:"ICMP Testing Rule"; sid:1000001; rev:1;)
> alert tcp any any -> any any (msg:"TCP Testing Rule"; sid:1000002; rev:1;)
> alert udp any any -> any any (msg:"UDP Testing Rule"; sid:1000003; rev:1;)
> alert icmp any any -> $HOME_NET any (msg:"ICMP Test"; sid:1000004; rev:1;)
> ```
> 
> Then clicked the 'Synchronize grid' button at the top of SOC and I started to get alerts for the new rules (after about 15 minutes for the grid sync to complete) ![image](https://private-user-images.githubusercontent.com/94730068/301302569-a1a3996b-29f9-4526-a035-7fe26e39cf4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MzYxMzEsIm5iZiI6MTcwNjczNTgzMSwicGF0aCI6Ii85NDczMDA2OC8zMDEzMDI1NjktYTFhMzk5NmItMjlmOS00NTI2LWEwMzUtN2ZlMjZlMzljZjRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDIxMTcxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwYjU1OWNmZTM2YzliYTQwYzgxNjEzMGZhODdhMTVjZDkxMmNiOWRhMDAwMDNiYmE3NDFmMmQwZTU3NTZkZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SE8NcA5G87Qv-ChFAJp-DMqopp5QGUrapaYBA6wHCDI)

Looks like I wasn't paying attention. Indeed, after the fix everything worked. Although I am a little surprised that because of one rule the whole chain of rules does not work, as I understand it. But nevertheless, Onion works properly after the fix.
Thanks a lot!
